test/v2_integration.py

```diff
@@ -903,6 +907,112 @@ def test_http_multiva_threshold_fail():
+        command=["mysql",
+            "-h", "bmysql",
+            "-u", "root",
+            "--password=",
+            "-e", query,
+            "boulder_sa_integration",
+            ]
+        subprocess.check_call(command, shell=False, stderr=subprocess.STDOUT)
+
+    def flip_ids(oldID, newID):
+        """
+        flip_ids changes a registrations ID from one value to another. Note that
+        in order for this to succeed all other tables with foreign key
+        constraints on the registration.ID field must be adjusted or otherwise
+        dealt with.
+        """
+        run_query("UPDATE registrations SET id={0} WHERE id={1}".format(newID, oldID))
+
+    def remove_certs(newID):
+        """
+        remove_certs deletes rows created while the account registration ID was changed.
+        We need to delete these rows so that the original account registration
+        ID can be restored without violating the foreign key constraints on the
+        certs, precerts and serials tables. Updating the registrationID of these
+        rows first is difficult because the new value (the original ID) doesn't
+        exist in the registrations table yet. The best solution would be doing
+        all of this in one transaction but we're already deep in hacky
+        integration test yak shaving at this point...
+        """
+        run_query("DELETE FROM certificates WHERE registrationID={0}".format(newID))
+        run_query("DELETE FROM precertificates WHERE registrationID={0}".format(newID))
+        run_query("DELETE FROM serials WHERE registrationID={0}".format(newID))
+
+    # Update the account ID in the database to match an ID that exists in
+    # `test/example-multi-va-policy.yaml`'s disabledAccounts list. We do this
+    # with direct DB access because the alternative is hackish rewriting of the
+    # policy YAML file at runtime (especially since the reload event can't be
+    # easily detected). This approach is _also_ hackish, but marginally less so.
+    newID=99991337
+    flip_ids(acctID, newID)
+
+    # Update the in-memory account ID for the client instance to match
+    client.net.account = RegistrationResource(
+            body=client.net.account.body,
+            uri=acctURI.replace(acctID, str(newID)),
+            terms_of_service=client.net.account.terms_of_service)
+
+    # Configure a guestlist that will fail the multiVA threshold test by
+    # only allowing the primary VA.
+    guestlist = {"boulder": 1}
+
+    # Setup for a random domain name
+    domain, cleanup = multiva_setup(client, guestlist, domain=None)
+
+    # We do not expect any errors, even though the guestlist ensured multi-va
+    # failures, because the client was set up with an account key corresponding
+    # to a multi VA policy disabledAccount ID.
+    try:
+        chisel2.auth_and_issue([domain], client=client, chall_type="http-01")
+    finally:
+        cleanup()
+        remove_certs(newID)
+        flip_ids(newID, acctID)
```

Comment:
Why do we need to flip the accountID back when it's done? I assume this is so that subsequent runs don't run into a duplicate key problem trying to flip to the "special" accountID. It would be valuable to document that. Also, I think if that's the case, we need everything from the first `flip_ids` onwards to be wrapped in a try/finally, or failures during multiva_setup or account creation could leave things in a bad state. Alternately I would be fine with catching the duplicate key problem when trying to flip to the "special" accountID and giving instructions to fix up the local DB.

Reply:
> Why do we need to flip the accountID back when it's done? I assume this is so that subsequent runs don't run into a duplicate key problem trying to flip to the "special" accountID. It would be valuable to document that

Yup, you're assuming correctly. I added some comments documenting this more explicitly in 
463c4d7.

>  Also, I think if that's the case, we need everything from the first flip_ids onwards to be wrapped in a try/finally, or failures during multiva_setup or account creation could leave things in a bad state

Good call. I moved more into the try/finally block in 8cb26ce.